src/workingtitle-vcockpits-instruments-navsystems/html_ui/Pages/VCockpit/Instruments/NavSystems/Shared/ConfigHandler.js

```diff
@@ -0,0 +1,103 @@
+class ConfigLoader {
+    // This class kind of became callback hell without me meaning for it to.
+    // TODO:  make it less of a callback hell
+    constructor(basepath) {
+        let vfspath = "/VFS/" + basepath.replace(/\\/g, "/");
+        let vfspathParts = vfspath.split("/");
+        this._vfspath = vfspathParts.slice(0, -2).join("/")
+    }
+
+    loadCfg(filename, cb) {
```

Comment:
If you wanted this to be slightly less callback madness, you could use promises here:

```javascript
return new Promise((resolve, reject) => {
  Utils.loadFile(`${this._vfspath}/${filename}`, (text) => {
    let parser = new CfgParser()
    let out = parser.parse(text);
    resolve(out);
  });
});
```

Don't forget semicolons, have a few missing after your `loadFile` calls in this class. ;-)

Reply:
Thanks.   I tried mucking with something along those lines but couldn't get it sorted.  I could give it another go though.  You have more time in the javascript mindset than I do;  for this environment do we gain enough by using promises instead of callbacks?    Callbacks fit better into my mindspace but I don't want to write pointlessly ugly code and async JS is pretty new to me.

Reply:
(Just to follow up for the Permanent Record(tm),  we discussed this on the side and I'll spend a little time reworking this with promises.)